qualification/report/generate_pdf.py

```diff
@@ -391,18 +391,24 @@ def __init__(self, results: Sequence[Result]) -> None:
         self.group = self.results[0].group
         for result in self.results:
             assert result.base_nodeid == self.base_nodeid, "base_nodeids do not match"
-            if result.text_name != self.text_name:
-                raise ValueError(
-                    f"Inconsistent text names for tests in ResultSet ({result.text_name} != {self.text_name})"
-                )
-            if result.blurb != self.blurb:
-                raise ValueError(f"Inconsistent blurbs for tests in ResultSet {self.base_nodeid}")
+            self.text_name = self._merge(self.text_name, result.text_name, "text names")
+            self.blurb = self._merge(self.blurb, result.blurb, "blurbs")
             self.duration += result.duration
             self.outcome_counts[result.outcome] += 1
             for requirement in result.requirements:
                 if requirement not in self.requirements:
                     self.requirements.append(requirement)
 
+    def _merge(self, a: str, b: str, name: str) -> str:
+        """Combine a property from two results.
+
+        If either is blank, the other is used. If both are non-blank and they
+        don't match, raise a :exc:`ValueError`.
+        """
+        if a != b and a and b:
+            raise ValueError(f"Inconsistent {name} for tests in ResultSet {self.base_nodeid}")
+        return a or b
```

Comment:
This is a rather weird syntax choice. When did Python start returning a string in this kind of case?

I'd have expected a boolean return given what's here.

Reply:
It's been like that as far as I can recall. See https://docs.python.org/3/reference/expressions.html#boolean-operations. Mostly it's an annoyance (e.g. it's the reason you needed to wrap `bool(...)` around a numpy boolean expression, to force the result of an `and` to be a Python bool), but in this case it's actually pretty handy (the link above describes exactly this use case, so I feel justified in saying it's Pythonic :-).

Fun fact: Python didn't originally have the ternary syntax ("x if cond else y") and the idiomatic way to do it was `cond and x or y` although if `x` was falsy it wouldn't work.